src/unicode.cpp

```diff
@@ -262,10 +262,8 @@ constexpr static uint8_t is_forbidden_domain_code_point_table[] = {
   }
 
   unsigned constexpr convert_hex_to_binary(const char c) noexcept {
-    // this code can be optimized.
-    if (c <= '9') { return c - '0'; }
-    char del = c >= 'a' ? 'a' : 'A';
-    return 10 + (c - del);
+    if (c <= '9') return c - '0';
```

Comment:
```suggestion
    if (c <= '9') return { c - '0'; }
```

Reply:
If hex-to-binary is ever a bottleneck, there are much faster ways to deal with it. At this point in time, I suspect that it is rarely a bottleneck.